common/src/main/java/tachyon/worker/DataServerMessage.java

```diff
@@ -224,6 +220,12 @@ private void generateHeader() {
     mHeader.putLong(mOffset);
     mHeader.putLong(mLength);
 
+    if (mMessageType == RPCMessage.Type.RPC_BLOCK_READ_REQUEST) {
+      // The request message has a lockId and a sessionId
+      mHeader.putLong(mLockId);
+      mHeader.putLong(mSessionId);
+    }
```

Comment:
We should either use `else if` with the following if statement, or `switch` on the enum